### DISCUSSION TITLE
Improvements for links in v4

### DISCUSSION THREAD

**Discussion by mikekistler** [OAI/sig-moonwalk#58] [Ideas]
I've put together a proposal for reworking/extending support for links in OASv4/Moonwalk.

https://gist.github.com/mikekistler/1983a8249c0df56a46f94b4df1b7181d

I'd be happy for feedback on this.

**Comment by ahl** [comment]
What is the purpose of links? How are they currently used, and how would they be ideally be used if shortcomings were removed?

I have seen them as interesting documentation addenda, but nothing that couldn't be equally well-expressed in, say, the description field for an operation.

**Reply by graipher** [reply]
Links are machine-readable, in contrast to the description which at best would use some convention.

Links are used by tools such as 
* `graphql-mesh` to generate & run GraphQL (https://github.com/OAI/OpenAPI-Specification/issues/1327#issuecomment-772904764) 
* `schemathesis` to perform automated property-based tests (https://schemathesis.readthedocs.io/en/stable/stateful.html)
* ..?

**Comment by handrews** [comment]
Also of interest (closed in the other repo as the discussion should continue here, please feel free to copy over any points of particular interest):

* OAI/OpenAPI-Specification#601
* OAI/OpenAPI-Specification#783
* OAI/OpenAPI-Specification#1011
* OAI/OpenAPI-Specification#1315
* OAI/OpenAPI-Specification#1327
* OAI/OpenAPI-Specification#1329
* OAI/OpenAPI-Specification#1593
* OAI/OpenAPI-Specification#1594
* OAI/OpenAPI-Specification#2122
* OAI/OpenAPI-Specification#2196
* OAI/OpenAPI-Specification#3228
